src/packages/PreRecordedClient.ts

```diff
@@ -0,0 +1,174 @@
+
+      return await this._makeSyncPrerecordedRequest(options, endpoint, body);
+    } catch (error) {
+      if (isDeepgramError(error)) {
+        return { result: null, error };
+      }
+
+      throw error;
+    }
+  }
+
+  async transcribeFile(
+    source: FileSource,
+    options?: PrerecordedOptions,
+    endpoint = "v1/listen"
+  ): Promise<DeepgramResponse<SyncPrerecordedResponse>> {
+    try {
+      this._setFileMimetypeHeaders(source);
+
+      const body = this._getPrerecordedFileBody(source);
+
+      return await this._makeSyncPrerecordedRequest(options, endpoint, body);
+    } catch (error) {
+      if (isDeepgramError(error)) {
+        return { result: null, error };
+      }
+
+      throw error;
+    }
+  }
+
+  async transcribeUrlCallback(
+    source: UrlSource,
+    callback: string,
+    options?: PrerecordedOptions,
+    endpoint = "v1/listen"
+  ): Promise<DeepgramResponse<AsyncPrerecordedResponse>> {
+    try {
+      const body = this._getPrerecordedUrlBody(source);
+
+      return await this._makeAsyncPrerecordedRequest(options, callback, endpoint, body);
+    } catch (error) {
+      if (isDeepgramError(error)) {
+        return { result: null, error };
+      }
+
+      throw error;
+    }
+  }
+
+  async transcribeFileCallback(
+    source: FileSource,
+    callback: string,
+    options?: PrerecordedOptions,
+    endpoint = "v1/listen"
+  ): Promise<DeepgramResponse<AsyncPrerecordedResponse>> {
+    try {
+      this._setFileMimetypeHeaders(source);
+
+      const body = this._getPrerecordedFileBody(source);
+
+      return await this._makeAsyncPrerecordedRequest(options, callback, endpoint, body);
+    } catch (error) {
+      if (isDeepgramError(error)) {
+        return { result: null, error };
+      }
+
+      throw error;
+    }
+  }
+
+  private _setFileMimetypeHeaders(source: FileSource): void {
+    if (source.mimetype === undefined || source.mimetype.length === 0) {
+      throw new DeepgramError("Mimetype must be provided if the source is a Buffer or a Readable");
+    }
+
+    this.headers["Content-Type"] = source.mimetype;
+  }
+
+  private _getPrerecordedUrlBody(source: UrlSource): string {
+    let body;
+    if (isUrlSource(source)) {
+      body = JSON.stringify(source);
+    } else {
+      throw new DeepgramError("Unknown transcription source type");
+    }
+
+    return body;
+  }
+
+  private _getPrerecordedFileBody(source: FileSource): Buffer | Readable {
+    let body;
+    if (isBufferSource(source)) {
+      body = source.buffer;
+    } else if (isReadStreamSource(source)) {
+      body = source.stream;
+    } else {
+      throw new DeepgramError("Unknown transcription source type");
+    }
+
+    return body;
+  }
+
+  private async _makeSyncPrerecordedRequest(
+    options: PrerecordedOptions | undefined,
+    endpoint: string,
+    body: string | Buffer | Readable
+  ): Promise<DeepgramResponse<SyncPrerecordedResponse>> {
+    if (options !== undefined && "callback" in options) {
```

Comment:
It's possible they would add a callback to the request like this:

```
deepgram.listen.prerecorded.transcribeUrl(audio_url, options, callback)
```

This wouldn't cause the error to throw because it would assume the callback url is the endpoint, right? So it would show them a 400 error instead of this one.

Reply:
We're going to accept an object as the callback URL